crates/nargo/src/cli/prove_cmd.rs

```diff
@@ -99,9 +99,8 @@ pub fn compile_circuit_and_witness<P: AsRef<Path>>(
         )
     }
 
-    let abi = compiled_program.abi.unwrap();
-
-    let mut solved_witness = process_abi_with_input(abi, witness_map)?;
+    let abi = compiled_program.abi.as_ref().unwrap();
+    let mut solved_witness = process_abi_with_input(abi.clone(), witness_map)?;
```

Comment:
Why do we clone the abi now? Looks like it isn't used below so we should be able to move it

Reply:
I think this came from a merge to master due to other changes. It seems if we take away the `as_ref` on `let abi = compiled_program.abi.as_ref().unwrap();` there will be a partial move and we will not be able to return `compiled_program.` Leaving the `as_ref` we must do `abi.clone()` as `abi` is `&Abi` while `process_abi_with_input` takes in an `Abi` type